env/staging/create_metrics_lambda/lambda/create_metric.js

```diff
@@ -21,31 +21,69 @@ exports.handler = async (event, context) => {
     // expire after 24 hours
     const ttl = (Math.floor(Date.now()/1000) + 86400).toString();
 
-    const params = {
-        TableName: process.env.TABLE_NAME,
-        Item : {
-            "uuid": {
-                S: uuidv4(),
-            },
-            "expdate" : {
-                N: ttl,
-            },
-            "raw": {
-                S: event.body,
-            },
-        },
-    };
-
+    const eventBody = event.body;
+    
     try {
+        // The maximum item size in DynamoDB is 400 KB
+        const payloadLength = new TextEncoder().encode(event.body.payload).length;
+        if (payloadLength > process.env.SPLIT_THRESHOLD){
+            console.info(`Large payload being split; size: ${payloadLength}`);
+            const results = splitPayload(event.body.payload);
 
-        await dynamodb.putItem(params).promise();
+            results.forEach((result) => {
+                result.forEach((r) => {
```

Comment:
Is this going to do what we want? If the pay load is too big by say 50 bytes, then splitPayload splits into `[[first], [second]]`. This second array will loop over each element of `[first]` and `[second]` and insert as individual records, won't it?

Reply:
The payload split function returns an array in the below format. Found this the easiest way with it being recursive and minimizing array manipulation.

[[[{
    "region": "",
    "timestamp": 1621957249091,
    "identifier": "star1"
},{
    "region": "",
    "timestamp": 1621957249096,
    "identifier": "moon2"
},{
    "region": "",
    "timestamp": 1621957249091,
    "identifier": "star3"
},{
    "region": "",
    "timestamp": 1621957249096,
    "identifier": "moon4"
},{
    "region": "",
    "timestamp": 1621957249091,
    "identifier": "star5"
},{
    "region": "",
    "timestamp": 1621957249096,
    "identifier": "moon6"
}],[{
    "region": "",
    "timestamp": 1621957249091,
    "identifier": "star7"
},{
    "region": "",
    "timestamp": 1621957249096,
    "identifier": "moon8"
},{
    "region": "",
    "timestamp": 1621957249091,
    "identifier": "star9"
},{
    "region": "",
    "timestamp": 1621957249096,
    "identifier": "moon10"
},{
    "region": "",
    "timestamp": 1621957249091,
    "identifier": "star11"
},{
    "region": "",
    "timestamp": 1621957249096,
    "identifier": "moon12"
}]],[[{
    "region": "",
    "timestamp": 1621957249091,
    "identifier": "star13"
},{
    "region": "",
    "timestamp": 1621957249096,
    "identifier": "moon14"
},{
    "region": "",
    "timestamp": 1621957249091,
    "identifier": "star15"
},{
    "region": "",
    "timestamp": 1621957249096,
    "identifier": "moon16"
},{
    "region": "",
    "timestamp": 1621957249091,
    "identifier": "star17"
},{
    "region": "",
    "timestamp": 1621957249096,
    "identifier": "moon18"
}],[{
    "region": "",
    "timestamp": 1621957249091,
    "identifier": "star19"
},{
    "region": "",
    "timestamp": 1621957249096,
    "identifier": "moon20"
},{
    "region": "",
    "timestamp": 1621957249091,
    "identifier": "star21"
},{
    "region": "",
    "timestamp": 1621957249096,
    "identifier": "moon22"
},{
    "region": "",
    "timestamp": 1621957249091,
    "identifier": "star23"
},{
    "region": "",
    "timestamp": 1621957249096,
    "identifier": "moon24"
}]]]

Reply:
That will depend on how much it recurses does it not? Each time it recurses it will push the `[[], []]` structure on the parent list.

Reply:
I'll reduce the split size and test to validate. i think you're onto something.

Reply:
You're 100% right Dan. Going to fix and compare the performance with the non-recursive example you provided below.

Reply:
recursive won the maintenance and speed race, so old implementation is gone.